v2/paths.go

```diff
@@ -37,3 +38,23 @@ func PidGroupPath(pid int) (string, error) {
 	p := fmt.Sprintf("/proc/%d/cgroup", pid)
 	return parseCgroupFile(p)
 }
+
+// VerifyGroupPath verifies the format of group path string g.
+// The format is same as the third field in /proc/PID/cgroup.
+// e.g. "/user.slice/user-1001.slice/session-1.scope"
+//
+// g must be a "clean" absolute path starts with "/", and must not contain "/sys/fs/cgroup" prefix.
+//
+// VerifyGroupPath doesn't verify whether g actually exists on the system.
+func VerifyGroupPath(g string) error {
+	if !strings.HasPrefix(g, "/") {
+		return ErrInvalidGroupPath
+	}
+	if filepath.Clean(g) != g {
+		return ErrInvalidGroupPath
+	}
+	if strings.HasPrefix(g, "/sys/fs/cgroup") {
+		return ErrInvalidGroupPath
+	}
+	return nil
+}
```

Comment:
Since we return here same error for all cases, don't you want to use something like:
```
func VerifyGroupPath(g string) error {
	if !strings.HasPrefix(g, "/") ||
	   filepath.Clean(g) != g ||
           strings.HasPrefix(g, "/sys/fs/cgroup") {
		return ErrInvalidGroupPath
	}

	return nil
}
```

Reply:
I feel it is less readable